backend/frontend/views.py

```diff
@@ -368,16 +373,19 @@ def action_start_card(request, card_id):
 
 
 @user_passes_test_or_forbidden(can_view_user_board)
-def course_component_details(request, project_id):
-    project = get_object_or_404(RecruitProject, id=project_id)
+def course_component_details(request, id):
+    request_url = request.build_absolute_uri()
+    type = request_url.split("/")[3]
 
-    board_status = [
-        value
-        for key, value in AgileCard.STATUS_CHOICES
-        if key == project.agile_card_status
-    ][0]
+    if type == "topic":
```

Comment:
This is a bit messy. And it also leaves space for typos and nonesensce down the road because urls chance. We should keep our url code cohesive. This cunction should not know about what the url structure is. 

We have 2 options:

1. change the url to something like this "progress_details/<content_type>/<id>"
2. make use of the resolve function to get the view name 

I think this will work:

```
   from django.core.urlresolvers import resolve
   current_url = resolve(request.path_info).url_name
```

That said, I think option 1 is better. Option 2 is brittle because it assumes that the view name is unlikely to change. We might want to change it because `project` and `topic` are not very descriptive in that context.

Reply:
If you are doing something that feels like a hack, it's probably a hack. 

Always think about the things that the code knows about. The knowledge of the code. What will happen if something changes? Is it likely to change?

Try to see the Django way to do things. Try to understand why best practices are in place. Why do we not hardcode urls? Since we don't hardcode urls why would you think it's ok to write code that expects a certain url structure? Does this not cause exactly the same problems as a hardcoded url? 

It's like you coded this thing with a hammer. You're working around the system in order to force things to work. Work with the framework not against it.